vigor/views.py

```diff
@@ -4,30 +4,45 @@
 from .forms import UserCreationFormExtended
 from django.contrib.auth import authenticate
 from django.contrib.auth import login as auth_login
-
-
 from django.views.decorators.csrf import csrf_exempt
+from .models import User
 
 def index(request):
     template = loader.get_template("vigor/index.html")
     context = {}
     return HttpResponse(template.render(context, request))
 
 def login(request):
-    return HttpResponse("Login Page.")
+    if request.user.is_authenticated():
+        return HttpResponseRedirect("/vigor/dashboard")
+
+    if request.method == 'POST':
+        username = request.POST['username']
+        password = request.POST['password']
+        user = authenticate(username=username, password=password)
+        if user is not None:
+            auth_login(request, user)
+            return HttpResponseRedirect("/vigor/dashboard") # change to just `/dashboard` when in production
+        else:
+            print("Internal Server Error")
+    return render(request, 'vigor/login.html')
 
 @csrf_exempt
 def signup(request):
+    if request.user.is_authenticated():
+        return HttpResponseRedirect("/vigor/dashboard")
+
     if request.method == "POST":
         form = UserCreationFormExtended(data = request.POST)
         if form.is_valid():
             form.save()
             username = form.cleaned_data["username"]
             password = form.cleaned_data["password1"]
             user = authenticate(username=username, password=password)
+            print(user)
             if user is not None:
                 auth_login(request, user)
-                return HttpResponseRedirect("/dashboard")
+                return HttpResponseRedirect("/vigor/dashboard") # change to just `/dashboard` when in production
```

Comment:
I think this was more for me to remember since I haven't exactly used Django before, but I will remove this and reference this whenever I forget.